File: frontend/src/utils/igv.js
const IGV = 0.18

export function calcularLinea(cantidad, precioUnitario, igvIncluido = false) {
  const cant = Number(cantidad) || 0
  const precio = Number(precioUnitario) || 0
  if (igvIncluido) {
    const importe = round2(precio * cant)
    const subtotal = round2(importe / (1 + IGV))
    const igv = round2(importe - subtotal)
    return { subtotal, igv, importe }
  }
  const subtotal = round2(precio * cant)
  const igv = round2(subtotal * IGV)
  const importe = round2(subtotal + igv)
  return { subtotal, igv, importe }
}

export function sumarTotales(items) {
  return items.reduce(
    (acc, it) => {
      const { subtotal, igv, importe } = calcularLinea(it.cantidad, it.precio_unitario, it.igv_incluido)
      acc.subtotal += subtotal
      acc.igv += igv
      acc.importe += importe
      return acc
    },
    { subtotal: 0, igv: 0, importe: 0 },
  )
}

function round2(n) {
  return Math.round(n * 100) / 100
}

export function formatoMoneda(valor) {
  return Number(valor || 0).toLocaleString('es-PE', { minimumFractionDigits: 2, maximumFractionDigits: 2 })
}

File: backend/logistica/utils/igv.py
from decimal import Decimal, ROUND_HALF_UP

IGV_TASA = Decimal('0.18')
IGV_FACTOR = Decimal('1.18')


def calcular_linea(cantidad, precio_unitario, igv_incluido=False):
    """Calcula subtotal (sin IGV), IGV e importe total de una línea."""
    cantidad = Decimal(str(cantidad or 0))
    precio = Decimal(str(precio_unitario or 0))
    if igv_incluido:
        importe_total = (precio * cantidad).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
        subtotal = (importe_total / IGV_FACTOR).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
        igv = (importe_total - subtotal).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
    else:
        subtotal = (precio * cantidad).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
        igv = (subtotal * IGV_TASA).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
        importe_total = (subtotal + igv).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
    return subtotal, igv, importe_total
